audioswitch/src/test/java/com/twilio/audioswitch/selection/AudioDeviceSelectorTest.kt

```diff
@@ -348,9 +349,16 @@ class AudioDeviceSelectorTest {
         whenever(buildWrapper.getVersion()).thenReturn(Build.VERSION_CODES.N_MR1)
         audioDeviceSelector.start(audioDeviceChangeListener)
         audioDeviceSelector.activate()
+        val audioFocusListenerCaptor =
```

Comment:
The verifications look great here @tejas-n ! One small change though. Can you please move the `requestAudioFocus` verify to the `activate should set audio focus using pre Android O method if api version is 25` test case and leave the `abandonAudioFocus` verify in the `deactivate should abandon audio focus using pre Android O method if api version is 25` test case? This will make the tests more granular and be more consistent with the other test cases. Thanks!

Reply:
Correct me if I'm wrong but wouldn't we need the `verify()` in this method to capture the audio focus listener which was used to request audio focus and compare it to the one which was used for abandoning?

If we move that to `activate should set audio focus using pre Android O method if api version is 25`, this particular test wouldn't have access to the captured audio focus listener, right?

Reply:
Ah good point cause you are using a capturer. How about instead of using a capturer, let's inject the `OnAudioFocusChangeListener` into the constructor. You can then easily inject a mock into the constructor similar to what was done for post Oreo device test cases with the `AudioFocusRequestWrapper`. I went ahead and made this change in separate commit that you can find [here](https://github.com/twilio/audioswitch/commit/1aaf8a061b36c28dc54b08ca163863764177629a). Thanks!

Reply:
Thanks, @Alton09, that looks good. I've merged that commit to this branch.

Reply:
Thanks! We'll work on getting this PR merged in soon. Thanks again for this bug fix @tejas-n!

Reply:
Cheers!